File: backend/app/Models/SolicitudCompra.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class SolicitudCompra extends Model
{
    protected $table = 'solicitudes_compra';

    protected $fillable = [
        'codigo',
        'fecha_solicitud',
        'usuario_solicita_id',
        'usuario_aprueba_id',
        'estado',
        'observaciones',
    ];

    protected function casts(): array
    {
        return [
            'fecha_solicitud' => 'datetime',
        ];
    }

    public function usuarioSolicita()
    {
        return $this->belongsTo(User::class, 'usuario_solicita_id');
    }

    public function usuarioAprueba()
    {
        return $this->belongsTo(User::class, 'usuario_aprueba_id');
    }

    public function detalles()
    {
        return $this->hasMany(SolicitudCompraDetalle::class, 'solicitud_id');
    }

    public function ordenesCompra()
    {
        return $this->hasMany(OrdenCompra::class);
    }
}

File: app_abarrotes/lib/models/solicitud_compra.dart
class SolicitudCompra {
  final int id;
  final String codigo;
  final String fechaSolicitud;
  final String estado;
  final String? observaciones;

  SolicitudCompra({
    required this.id,
    required this.codigo,
    required this.fechaSolicitud,
    this.estado = 'pendiente',
    this.observaciones,
  });

  factory SolicitudCompra.fromJson(Map<String, dynamic> json) {
    return SolicitudCompra(
      id: json['id'],
      codigo: json['codigo'],
      fechaSolicitud: json['fecha_solicitud'],
      estado: json['estado'] ?? 'pendiente',
      observaciones: json['observaciones'],
    );
  }

  Map<String, dynamic> toJson() => {
    'codigo': codigo,
    'fecha_solicitud': fechaSolicitud,
    'estado': estado,
    'observaciones': observaciones,
  };
}
